src/ResourceManager/SignalR/Commands.SignalR/Cmdlets/NewAzureRmSignalR.cs

```diff
@@ -0,0 +1,139 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Common.Strategies;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Commands.SignalR.Models;
+using Microsoft.Azure.Commands.SignalR.Strategies;
+using Microsoft.Azure.Management.SignalR.Models;
+using System.Collections.Generic;
+using System.Management.Automation;
+using System.Threading.Tasks;
+using Microsoft.Azure.Commands.SignalR.Strategies.ResourceManager;
+using Microsoft.Azure.Commands.SignalR.Strategies.SignalRRp;
+
+namespace Microsoft.Azure.Commands.SignalR.Cmdlets
+{
+    [Cmdlet(VerbsCommon.New, SignalRNoun, SupportsShouldProcess = true)]
+    [OutputType(typeof(PSSignalRResource))]
+    public sealed class NewAzureRmSignalR : SignalRCmdletBase
+    {
+        private const string DefaultSku = "Basic_DS2";
+
+        [Parameter(
+            Mandatory = false,
+            Position = 0,
+            HelpMessage = "The resource group name. The default one will be used if not specified.")]
+        [ValidateNotNullOrEmpty()]
+        public override string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            Position = 1,
+            HelpMessage = "The SignalR service name.")]
+        [ValidateNotNullOrEmpty()]
+        public string Name { get; set; }
+
```

Comment:
Position: if the parameter **Name**  is mandatory should it go before the **ResourceGroupName**  parameter?

Reply:
I wonder how often the users use `Set-AzureRmDefault -ResourceGroupName` in daily practice. Most of other services uses `ResourceGroupName` as the first positional parameter followed by other parameters. I would vote to keep it as it is to maintain consistent behavior among all the Azure services.

The user may use `New-AzureRmSignalR -Name` to pick up the default resource group.